src/elfpy/markets.py

```diff
@@ -151,185 +118,64 @@ def check_fees(
-                )
-                (
-                    without_fee_or_slippage,
-                    output_with_fee,
-                    output_without_fee,
-                    fee,
-                ) = trade_results
-                d_base_asset = -output_with_fee
-                d_token_asset = amount
-                d_base_asset_slippage = abs(without_fee_or_slippage - output_without_fee)
-                d_token_asset_slippage = 0
-                d_base_asset_fee = fee
-                d_token_asset_fee = 0
-                d_base_asset_orders = 1
-                d_token_asset_orders = 0
-                d_base_asset_volume = output_with_fee
-                d_token_asset_volume = 0
-            elif token_in == "base" and token_out == "fyt":
-                in_reserves = self.base_asset
-                out_reserves = self.token_asset + self.total_supply
-                trade_results = self.pricing_model.calc_out_given_in(
-                    amount,
-                    in_reserves,
-                    out_reserves,
-                    token_out,
-                    self.fee_percent,
-                    self.time_remaining,
-                    self.init_share_price,
-                    self.share_price,
-                )
-                (
-                    without_fee_or_slippage,
-                    output_with_fee,
-                    output_without_fee,
-                    fee,
-                ) = trade_results
-                d_base_asset = amount
-                d_token_asset = -output_with_fee
-                d_base_asset_slippage = 0
-                d_token_asset_slippage = abs(without_fee_or_slippage - output_without_fee)
-                d_base_asset_fee = 0
-                d_token_asset_fee = fee
-                d_base_asset_orders = 0
-                d_token_asset_orders = 1
-                d_base_asset_volume = 0
-                d_token_asset_volume = output_with_fee
+        # assign general trade details, irrespective of trade type
+        trade_details = user_action.copy()
+        trade_details["fee_percent"] = self.fee_percent
+        trade_details["init_share_price"] = self.init_share_price
+        trade_details["share_price"] = self.share_price
+        trade_details["share_reserves"] = self.share_reserves
+        trade_details["bond_reserves"] = self.bond_reserves
+        # for each position, specify how to forumulate trade and then execute
+        if user_action["action_type"] == "open_long":  # buy to open long
+            trade_details["direction"] = "out"  # calcOutGivenIn
+            trade_details["token_out"] = "pt"  # buy unknown PT with known base
+            market_deltas, wallet_deltas = self.pricing_model.open_long(trade_details)
+        elif user_action["action_type"] == "close_long":  # sell to close long
+            trade_details["direction"] = "out"  # calcOutGivenIn
+            trade_details["token_out"] = "base"  # buy unknown PT with known base
```

Comment:
I might be misunderstanding, but I think the comment should read
`# sell known PT for unknown base`